projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts

```diff
@@ -0,0 +1,2253 @@
+    deprecatedParams: [
+      {
+        className: 'ActionsSubject',
+        importPath: '@ngrx/store',
+      },
+      {
+        className: 'UserIdService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'MultiCartService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'SavedCartFacade',
+        importPath: '@spartacus/cart/saved-cart/root',
+      },
+      {
+        className: 'RoutingService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActionsSubject',
+        importPath: '@ngrx/store',
+      },
+      {
+        className: 'MultiCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ProductImportInfoService',
+        importPath: '@spartacus/cart/base/core',
+      },
+      {
+        className: 'MultiCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CartNotEmptyGuard',
+    importPath: '@spartacus/checkout/components',
+    deprecatedParams: [
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'SemanticPathService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'Router',
+        importPath: '@angular/router',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutAuthGuard',
+    importPath: '@spartacus/checkout/components',
+    deprecatedParams: [
+      {
+        className: 'AuthService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'AuthRedirectService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'CheckoutConfigService',
+        importPath: '@spartacus/checkout/components',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'SemanticPathService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'Router',
+        importPath: '@angular/router',
+      },
+      {
+        className: 'UserAccountFacade',
+        importPath: '@spartacus/user/account/root',
+      },
+      {
+        className: 'GlobalMessageService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'CheckoutDetailsService',
```

Comment:
This one is gone. All checkout stuff has to be checked by developers who did the checkout refactor.